monitoring/port.go

```diff
@@ -104,14 +107,20 @@ func (c *portChecker) checkProcess(proc process, reporter health.Reporter) bool
 		if r.Protocol != proc.socket.proto() {
 			continue
 		}
-		// ignore sockets in time-wait and closed states since they're going
-		// away soon
 		switch proc.socket.state() {
 		case TimeWait:
+			// ignore sockets in time-wait and closed states since they're going
+			// away soon
 			log.Debugf("Ignoring %v for program %q(pid=%v).", formatSocket(proc.socket), proc.name, proc.pid)
 			continue
 		}
-		if uint64(proc.localAddr().port) >= r.From && uint64(proc.localAddr().port) <= r.To {
+		localAddr := proc.localAddr().ip.String()
+		addr := r.ListenAddr
+		if addr == "" {
+			addr = localAddr
+		}
+		if uint64(proc.localAddr().port) >= r.From && uint64(proc.localAddr().port) <= r.To &&
```

Comment:
This is more of a nitpick, but I think the readability of this condition would greatly improve if you made a couple of helper functions like:

```go
func isPortConflict(proc <type-of-proc>, range <type-of-r>) bool {}
func isIPConflict(proc <type-of-proc>, range <type-of-r>) bool {}

if isPortConflict(proc, r) && isIPConflict(proc, r) {}
```

And the code above that determines addr could also move to isIPConflict function.